.github/workflows/continuous-integration.yml

```diff
@@ -17,27 +17,23 @@ jobs:
         distribution: 'temurin'
         cache: maven
     - name: Checkout wala/IDE sources.
-      uses: actions/checkout@v3.5.2
-      with:
-        repository: wala/IDE
-        # fetch-depth: 50
-        path: ./IDE
+      run: git clone --depth=50 https://github.com/wala/IDE ${{ runner.temp }}/IDE
```

Comment:
Why not `--depth=1`?

Reply:
I don't see why not. This was also in the original Travis CI config. Travis CI clones a history of 50 the get the project sources. But, I found [this post](https://stackoverflow.com/questions/31278233/disadvantages-of-shallow-cloning-on-travis-and-other-ci-services#answer-53857834). Looks like we should stick with 50?

Reply:
I don't think the issue on that post is a concern for PR CI jobs.  If it comes up we can switch to 50, but for now I say depth 1.

Reply:
Are you sure? https://stackoverflow.com/questions/31278233/disadvantages-of-shallow-cloning-on-travis-and-other-ci-services#answer-53857834:~:text=Do%20not%20use%20a%20depth%3D1%20on%20your%20project%20for%20CI!

> Do not use a depth=1 on your project for CI!

Reply:
Well, I guess it was `1` before! https://github.com/actions/checkout#:~:text=%23%20Default%3A%201-,fetch%2Ddepth%3A%20%27%27,-%23%20Whether%20to

Reply:
Looks like there could be a problem with tags: https://github.com/actions/checkout/issues/217#issue-599945005.

We can try `1`. I don't foresee having a bunch of queued up builds.

Reply:
I also really like using the `concurrency` setting for GitHub Actions:

https://github.com/wala/WALA/blob/89de874efbc3369097e03b85493f4a3beb8e1a44/.github/workflows/continuous-integration.yml#L7-L9

This ensures there is only one build in progress for any PR or on the master branch.  If you push a new commit to the PR branch, it cancels the ongoing build before starting a new one.  If we want additional safety we can go with `depth=1` and also add this `concurrency` setting.

Reply:
Yeah, we have this kind of thing on Travis CI (there, it's specified on the web and not the YAML). The problem we've had with it is for builds emanating from tags. If you have a deploy action that only works on tagged commits, then you bump the version to the `SNAPSHOT` version, the CI stops building the tagged version (that was supposed to be deployed) and starts building the new `SNAPSHOT` version. The result is that the deployment gets skipped.

Reply:
If we ever need to do some kind of continuous deployment on GitHub Actions from tags, I'm sure we can configure it with a completely different `.yml` file, and the `concurrency` setting from one file shouldn't affect the other one.